### PR TITLE
Feature/5846 5849

### LINKED ISSUE #5846: Required participation rate and majority for success cannot be queried via APIs
We are missing API response data which satisfies the following AC

https://github.com/vegaprotocol/specs/blob/master/protocol/0028-GOVE-governance.md

If a new proposal is sucessfully submitted to the network (passing initial validation) the required participation rate and majority for success are defined and copied to the proposal and can be queried via APIs separately from the general network parameters. ([0028-GOVE-036](https://github.com/vegaprotocol/specs/blob/master/protocol/0028-GOVE-governance.md#0028-GOVE-036))

Currently the get proposal by ID or get proposals responses return this:-

```
{
  "data": {
    "proposal": {
      "id": "dba6e23f0d5f2aea37b2055634f3ae45cecd5eddbe4cbe6e9e1d418436b43d80",
      "reference": "D9F946C04E",
      "partyId": "f5e730252a1ac287beba2399fd4bddd91e5226cc650fa655d2a8721194032c13",
      "state": "STATE_DECLINED",
      "timestamp": "1660138869725096000",
      "terms": {
        "closingTimestamp": "1660138879",
        "enactmentTimestamp": "1660138881",
        "updateNetworkParameter": {
          "changes": {
            "key": "reward.staking.delegation.payoutDelay",
            "value": "0s"
          }
        },
        "validationTimestamp": "0"
      },
      "reason": "PROPOSAL_ERROR_PARTICIPATION_THRESHOLD_NOT_REACHED",
      "rationale": {
        "description": "Update network parameter",
        "hash": "",
        "url": ""
      },
      "errorDetails": ""
    },
    "yes": [],
    "no": [],
    "yesParty": {},
    "noParty": {}
  }
}
```

This may be impacting out ability to also test this AC

If a new proposal "P" is sucessfully submitted to the network (passing initial validation) the required participation rate and majority for success are defined and copied to the proposal. If an independent network parameter change proposal is enacted changing either required participation of majority then proposal "P" uses its own values for participation and majority; not the newly enacted ones. ([0028-GOVE-037](https://github.com/vegaprotocol/specs/blob/master/protocol/0028-GOVE-governance.md#0028-GOVE-037))

### LINKED ISSUE #5849: Proposals should ignore network param changes after submitting a proposal
Have a test which fails to satisfy the following AC

https://github.com/vegaprotocol/specs/blob/master/protocol/0028-GOVE-governance.md

If a new proposal "P" is sucessfully submitted to the network (passing initial validation) the required participation rate and majority for success are defined and copied to the proposal. If an independent network parameter change proposal is enacted changing either required participation of majority then proposal "P" uses its own values for participation and majority; not the newly enacted ones. ([0028-GOVE-037](https://github.com/vegaprotocol/specs/blob/master/protocol/0028-GOVE-governance.md#0028-GOVE-037))

test test_proposal_impacted_network_change performs the following

Run 1

- network restarts
- checks governance.proposal.market.requiredMajority is 0.66
- creates proposal
- submits market proposal votes with Y, Y, N
- waits for proposal enactment
- proposal is Enacted

Run 2

- network restarts
- checks governance.proposal.market.requiredMajority is 0.66
- creates proposal
- submits market proposal votes with Y, Y, N
- submits param change governance.proposal.market.requiredMajority to be "1" (100%) - this is enacted
- waits for proposal enactment
- proposal is Declined ("reason":"PROPOSAL_ERROR_MAJORITY_THRESHOLD_NOT_REACHED")

Expected the 2nd run to pass but it seems the proposal is using the requiredMajority setting after proposal is submitted at "Open" state.

### PR DESCRIPTION
**Pull request by jeremyletang**
close #5846 
close #5849

### CODE REVIEW

File: core/governance/engine.go [hunk 1]
```diff
@@ -713,6 +723,12 @@ func (e *Engine) validateOpenProposal(proposal *types.Proposal) (types.ProposalE
 		}
 	}
 
+	// assign all requirement to the proposal itself.
```
Comment (wwestgarth) [inline core/governance/engine.go:726]: Minor: Do we need to do this both at the start and at the end of `validateOpenProposal`?

File: core/governance/engine.go [hunk 2]
```diff
@@ -855,9 +871,7 @@ func (e *Engine) closeProposal(ctx context.Context, proposal *proposal) {
 		return
 	}
 
-	params := e.mustGetProposalParams(proposal)
```
Comment (wwestgarth) [inline core/governance/engine.go:858]: I think `e.mustGetProposalParams()` is unused now and can probably be deleted.

**Review by wwestgarth — DISMISSED** [review]

File: datanode/entities/proposal.go [hunk 3]
```diff
@@ -38,29 +39,46 @@ type _Proposal struct{}
...
 }
 
 func (p *Proposal) ToProto() *vega.Proposal {
+	var lpMajority *string
+	if !p.RequiredLPMajority.IsZero() {
+		lpMajority = toPtr(p.RequiredLPMajority.String())
+	}
+	var lpParticipation *string
+	if !p.RequiredLPParticipation.IsZero() {
+		lpMajority = toPtr(p.RequiredLPParticipation.String())
```
Comment (wwestgarth) [inline datanode/entities/proposal.go:65]: This should be `lpParticipation =`.

File: core/types/governance_proposal.go [hunk 4]
```diff
@@ -268,15 +277,29 @@ func (p Proposal) IntoProto() *vegapb.Proposal {
...
 		terms = p.Terms.IntoProto()
 	}
+
+	var lpMajority *string
+	if !p.RequiredLPMajority.IsZero() {
+		lpMajority = toPtr(p.RequiredLPMajority.String())
+	}
+	var lpParticipation *string
+	if !p.RequiredLPParticipation.IsZero() {
+		lpMajority = toPtr(p.RequiredLPParticipation.String())
```
Comment (wwestgarth) [inline core/types/governance_proposal.go:287]: This should be `lpParticipation =`

File: core/governance/engine.go (hunk 1 above)
Comment (jeremyletang) [reply core/governance/engine.go:726]: not at all, only at the start. it's a mistake

**Review by wwestgarth — DISMISSED** [review]

**Review by ze97286 — DISMISSED** [review]

**Review by mattrussell36 — DISMISSED** [review]

**Review by ValentinTrinque — DISMISSED** [review]

**Review by ValentinTrinque — APPROVED** [review]

**Review by dexturr — APPROVED** [review]

**Pull request merged**